### PR TITLE
242 & 248 fix widget bugs

### PR DESCRIPTION
**Pull request by Orta21**
Ticket: 242 & 248

### Description

Update the widget bugs

### Release Plan

Asap

### CODE REVIEW

**Comment by Orta21** [comment]
Recording of changes

https://user-images.githubusercontent.com/36802074/211160465-ba3a7465-8258-44ae-80d0-79c5faf27f28.mov

File: connect-widget/app/src/shared/localStorage/custom-color.ts [hunk 1]
```diff
@@ -1,10 +1,13 @@
 const colorLocalStorageKey = "custom-color";
 
-export const getCustomColor = (paramColor: string | null) => {
+export const getCustomColor = (paramColor: string | null, isDemo: boolean) => {
   const color = validateColor(paramColor);
 
   if (color) {
-    localStorage.setItem(colorLocalStorageKey, color);
+    if (!isDemo) {
```
Comment (Goncharo) [inline connect-widget/app/src/shared/localStorage/custom-color.ts:7]: should this be `if (!isDemo && color)` ?

File: connect-widget/app/src/shared/localStorage/providers.ts [hunk 2]
```diff
@@ -12,11 +12,14 @@ const providers: DefaultProvider[] = [
 const providersLocalStorageKey = "providers";
 
 
-export const getProviders = (searchProviders: string | null): DefaultProvider[] => {
+export const getProviders = (searchProviders: string | null, isDemo: boolean): DefaultProvider[] => {
   const validProviders = getValidProviders(searchProviders);
 
   if (validProviders) {
-    localStorage.setItem(providersLocalStorageKey, JSON.stringify(validProviders));
+    if (!isDemo) {
```
Comment (Goncharo) [inline connect-widget/app/src/shared/localStorage/providers.ts:19]: should this be `if (!isDemo && validProviders)` ?

**Review by Goncharo — CHANGES_REQUESTED** [review]
Mostly good, couple of logic comments, also I think the `apple.png` was an accidental addition

**Comment by Orta21** [comment]
Fixing merge conflict

File: connect-widget/app/src/shared/localStorage/custom-color.ts (hunk 1 above)
Comment (Orta21) [reply connect-widget/app/src/shared/localStorage/custom-color.ts:7]: I had it like that but then the color below wasnt getting returned

File: connect-widget/app/src/shared/localStorage/providers.ts (hunk 2 above)
Comment (Orta21) [reply connect-widget/app/src/shared/localStorage/providers.ts:19]: Same issue as the colors. I could create a separate if check specifically for the localstorage

File: connect-widget/app/src/shared/localStorage/custom-color.ts (hunk 1 above)
Comment (Goncharo) [reply connect-widget/app/src/shared/localStorage/custom-color.ts:7]: nvm that's my bad
I totally missed the `if (color) {` in the original PR

File: connect-widget/app/src/shared/localStorage/providers.ts (hunk 2 above)
Comment (Goncharo) [reply connect-widget/app/src/shared/localStorage/providers.ts:19]: ditto

**Review by Goncharo — APPROVED** [review]
🚢 🚢 🚢 🚢 🚢

**Pull request merged**